### PR TITLE
Add test for progress bar

### PR DESCRIPTION
**Pull request by kssenii**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by kssenii** [comment]
The test fails because https://github.com/ClickHouse/ClickHouse/pull/25545 is not merged yet.

**Comment by kssenii** [comment]
@Mergifyio update

**Comment by kssenii** [comment]
The only failure - `Functional stateless tests (ubsan) — fail: 1` - Timeout on wating for prompt because of `Cannot create file` exception...
```code
01921_test_progress_bar:                                                [ FAIL ] 121.02 sec. - return code 1
2021-06-22 01:41:17 Traceback (most recent call last):
2021-06-22 01:41:17   File "/usr/share/clickhouse-test/queries/0_stateless/helpers/uexpect.py", line 170, in read
2021-06-22 01:41:17     data += self.queue.get(timeout=timeleft)
2021-06-22 01:41:17   File "/usr/lib/python3.8/queue.py", line 178, in get
2021-06-22 01:41:17     raise Empty
2021-06-22 01:41:17 _queue.Empty
2021-06-22 01:41:17 
2021-06-22 01:41:17 During handling of the above exception, another exception occurred:
2021-06-22 01:41:17 
2021-06-22 01:41:17 Traceback (most recent call last):
2021-06-22 01:41:17   File "/usr/share/clickhouse-test/queries/0_stateless/helpers/uexpect.py", line 144, in expect
2021-06-22 01:41:17     data = self.read(timeout=timeleft, raise_exception=True)
2021-06-22 01:41:17   File "/usr/share/clickhouse-test/queries/0_stateless/helpers/uexpect.py", line 178, in read
2021-06-22 01:41:17     raise TimeoutError(timeout)
2021-06-22 01:41:17 uexpect.TimeoutError: Timeout 119.872s
2021-06-22 01:41:17 
2021-06-22 01:41:17 During handling of the above exception, another exception occurred:
2021-06-22 01:41:17 
2021-06-22 01:41:17 Traceback (most recent call last):
2021-06-22 01:41:17   File "/usr/share/clickhouse-test/queries/0_stateless/01921_test_progress_bar.py", line 16, in <module>
2021-06-22 01:41:17     client1.expect(prompt)
2021-06-22 01:41:17   File "/usr/share/clickhouse-test/queries/0_stateless/helpers/uexpect.py", line 151, in expect
2021-06-22 01:41:17     raise exception
2021-06-22 01:41:17 uexpect.ExpectTimeoutError: Timeout 120.000s for ':\\) ' buffer 'clickhouse client\r\n\x1b[0J\x1b[?25hClickHouse client version 21.7.1.7238.\r\nConnecting to localhost:9000 as user default.\r\nConnected to ClickHouse server version 21.7.1 revision 54449.\r\n\r\nCode: 586. DB::ErrnoException: Cannot create file: /root/.clickhouse-client-history, errno: 17, strerror: File exists\r\n\r\nroot@5d49a1a754cc:/# '
```

**Comment by kssenii** [comment]
@Mergifyio update

**Comment by tavplubix** [comment]
> The only failure - `Functional stateless tests (ubsan) — fail: 1` - Timeout on wating for prompt because of `Cannot create file` exception...

It fails because of TOCTOU issue here:
https://github.com/ClickHouse/ClickHouse/blob/bdc8fe1ca08d707099d1f0c36fa0d1aa5bbfeca9/programs/client/Client.cpp#L578-L579

Could you please fix it by the way?

File: tests/queries/0_stateless/01921_test_progress_bar.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+CURDIR = os.path.dirname(os.path.realpath(__file__))
+sys.path.insert(0, os.path.join(CURDIR, 'helpers'))
+
+from client import client, prompt, end_of_block
+
+log = None
+# uncomment the line below for debugging
+#log=sys.stdout
+
+with client(name='client1>', log=log) as client1, client(name='client2>', log=log) as client2:
```
Comment (tavplubix) [inline tests/queries/0_stateless/01921_test_progress_bar.py:15]: `client2` is never used (looks like a copy-paste from `*live_view*` tests)

Btw, it's possible to write such tests without wrappers on python, for example see `01520_client_print_query_id`, `01179_insert_values_semicolon` and other `*.expect` tests. But tests on python are ok, as to me.

File: programs/client/Client.cpp [hunk 2]
```diff
@@ -577,7 +577,18 @@ class Client : public Poco::Util::Application
             }
 
             if (!history_file.empty() && !fs::exists(history_file))
-                FS::createFile(history_file);
+            {
+                try
+                {
+                    FS::createFile(history_file);
+                }
+                catch (const fs::filesystem_error & e)
```
Comment (tavplubix) [inline programs/client/Client.cpp:585]: But it throws `DB::ErrnoException`, not `fs::filesystem_error`

**Review by tavplubix — APPROVED** [review]

**Pull request merged**